### PR TITLE
Option for non persistent fs watcher

### PR DESCRIPTION
**Pull request by furanzujin**
File watchers can be started either in persistent or non persistent mode. A node process will be blocked in because of the watching state if the watcher is the only thing still running in the program.
The option added avoids this behavior and stop the node process if the watcher is the only thing left running.

### CODE REVIEW

**Comment by furanzujin** [comment]
@runk, my branch has bee created from the last stable tag (2.6.0) if I must rebase on master please let me know.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "maxmind",
-  "version": "2.5.0",
+  "version": "2.6.0",
```
Comment (runk) [inline package.json]: Please revert it - normally I bump version during release - see `./scripts/release` file for details.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "istanbul": "^0.4.5",
     "mocha": "^5.0.5",
     "nyc": "^11.4.1",
-    "sinon": "^4.4.2"
+    "sinon": "^5.0.1"
```
Comment (runk) [inline package.json]: 💯

**Comment by furanzujin** [comment]
@runk ,
I have rebased on master. Would you need me to do anything else, please let me know.
If you need any assistance on the lib feel free to ask.

**Comment by runk** [comment]
Thanks, Sorry for being pain in the ass, but can you revert changes in package-lock.json too, since you aren't changing anything in the packags.json itself? Happy to merge thereafter

**Comment by furanzujin** [comment]
Hello @runk,
Well I could uncommit the package-lock file but if I do so, the "npm ci" command of your Jenkins test with node 10 would fail because because of the cipm that raises an error if package and package-lock are out of sync.

See output

> › npm ci
> npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
> npm ERR! 
> npm ERR! 
> npm ERR! Invalid: lock file's sinon@4.4.2 does not satisfy sinon@^5.0.1
> npm ERR! 
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/charles/.npm/_logs/2018-06-07T05_50_43_856Z-debug.log

I have had the error in my attempt for [PR #95 ](https://travis-ci.org/runk/node-maxmind/builds/386523213)
And greenkeeper had it too in its
[PR #96  ](https://travis-ci.org/runk/node-maxmind/builds/386895834)

**Comment by furanzujin** [comment]
Actually, I didn't modify the package.json file (as you can see in [my commit](https://github.com/runk/node-maxmind/pull/95/commits/6c1b9c6265b06c96b18e1741978d95e6ea8a893d)).
I did not bump sinon nor the version. The version in the package file is already [2.6.0 on master](https://github.com/runk/node-maxmind/blob/master/package.json) as it is the current release ;)
The bump of sinon has been commited by your green keeper on may the 2nd.

**Comment by runk** [comment]
Damn it, my bad 🤦‍♂️  Merging, will resolve package and lock file conflicts in next PR. Thanks for the improvement 💯

**Comment by runk** [comment]
@furanzujin v2.7.0 is out.

**Comment by furanzujin** [comment]
Great :)

Thank you.

**Pull request merged**